### ISSUE TITLE
Allow to exclude files and subfolders from deploying

### ISSUE THREAD

**Issue by jakimowb** [g-sherman/plugin_build_tool#5]
Not all files stored in the source code repository might be required to be in the deployed plugin, e.g. some resources compiled into the resource.py or svg vector files from which png icons are created.

It would be great if the pb_tool.cfg allows to exclude files and folder like in a .gitignore file (https://git-scm.com/docs/gitignore):

`
[files]
excluded_files:
   *.svg
   *.pyc
   .hidden/trees/
   visible/trees/
`

**Comment by g-sherman** [comment]
pb_tool only copies files that are specified in the pb_tool.cfg file.

**Comment by jakimowb** [comment]
The `extra_dirs:` keyword allows to copy entire folder structures, This is a great, as it allows to use pb_tool also for larger projects, but it does not allow to exclude e.g. temporary files like `*.pyc`: 
```
extra_dirs:
    subproject_A
    subproject_B
    icons
```
In this case it becomes complicated to maintain a potentially long list of single file paths using the `"extras:"` keyword only (like the set of icons stored in folder `\icons`). That's why I think it would be very helpful to support a wildcard syntax to exclude or include specific files or file types.

**Comment by alessioarena** [comment]
Definitely a thumb up for this request.
If you have a git repository in a subfolder that you need to deploy with the plugin, this often causes "Access is denied" errors and they are not needed anyway

**Comment by Kohlroulade** [comment]
Yeah, I also have a project that contains many sub-folders, e.g. for images and static data as well as structured source-code. However there are plenty of files in the repository that aren't needed on production, but only for the build-process and thus shouldn't go to the build-artifact. I'd rather have the ability to exclude or include files by pattern also in the `pythonfiles`-section, in order to deploy an entire sub-folder containing python-files (but potentially ignoring any `__pycache__ ` for instance).
